File: MiniDOMOffice/CRM/BlackList/BlackListAddressCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Imports minidom.Anagrafica

Partial Public Class CustomerCalls
     
    Public Class BlackListAddressCursor
        Inherits DBObjectCursorBase(Of BlackListAddress)

        Private m_TipoRegola As New CCursorField(Of BlackListType)("Tipo")
        Private m_TipoContatto As New CCursorFieldObj(Of String)("TipoContatto")
        Private m_ValoreContatto As New CCursorFieldObj(Of String)("Valore")
        Private m_DataBlocco As New CCursorField(Of Date)("DataBlocco")
        Private m_IDBloccatoDa As New CCursorField(Of Integer)("IDBloccatoDa")
        Private m_NomeBloccatoDa As New CCursorFieldObj(Of String)("NomeBloccatoDa")
        Private m_MotivoBlocco As New CCursorFieldObj(Of String)("MotivoBlocco")

        Public Sub New()
        End Sub

        Public ReadOnly Property TipoRegola As CCursorField(Of BlackListType)
            Get
                Return Me.m_TipoRegola
            End Get
        End Property

        Public ReadOnly Property ValoreContatto As CCursorFieldObj(Of String)
            Get
                Return Me.m_ValoreContatto
            End Get
        End Property

        Public ReadOnly Property TipoContatto As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoContatto
            End Get
        End Property

        Public ReadOnly Property DataBlocco As CCursorField(Of Date)
            Get
                Return Me.m_DataBlocco
            End Get
        End Property

        Public ReadOnly Property IDBloccatoDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDBloccatoDa
            End Get
        End Property

        Public ReadOnly Property NomeBloccatoDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeBloccatoDa
            End Get
        End Property

        Public ReadOnly Property MotivoBlocco As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoBlocco
            End Get
        End Property

        Protected Overrides Function GetConnection() As CDBConnection
            Return CRM.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return CustomerCalls.BlackListAdresses.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_BlackListAddress"
        End Function
    End Class



End Class

File: MiniDOMOffice/CRM/BlackList/BlackListAddressCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;
namespace minidom
{
    public partial class CustomerCalls
    {

        /// <summary>
        /// Cursore su oggetti di tipo <see cref="BlackListAddress"/>
        /// </summary>
        [Serializable]
        public class BlackListAddressCursor 
            : minidom.Databases.DBObjectCursorBase<BlackListAddress>
        {
            private DBCursorField<BlackListType> m_TipoRegola = new DBCursorField<BlackListType>("Tipo");
            private DBCursorStringField m_TipoContatto = new DBCursorStringField("TipoContatto");
            private DBCursorStringField m_ValoreContatto = new DBCursorStringField("Valore");
            private DBCursorField<DateTime> m_DataBlocco = new DBCursorField<DateTime>("DataBlocco");
            private DBCursorField<int> m_IDBloccatoDa = new DBCursorField<int>("IDBloccatoDa");
            private DBCursorStringField m_NomeBloccatoDa = new DBCursorStringField("NomeBloccatoDa");
            private DBCursorStringField m_MotivoBlocco = new DBCursorStringField("MotivoBlocco");

            /// <summary>
            /// Costruttore
            /// </summary>
            public BlackListAddressCursor()
            {
            }

            /// <summary>
            /// TipoRegola
            /// </summary>
            public DBCursorField<BlackListType> TipoRegola
            {
                get
                {
                    return m_TipoRegola;
                }
            }

            /// <summary>
            /// ValoreContatto
            /// </summary>
            public DBCursorStringField ValoreContatto
            {
                get
                {
                    return m_ValoreContatto;
                }
            }

            /// <summary>
            /// TipoContatto
            /// </summary>
            public DBCursorStringField TipoContatto
            {
                get
                {
                    return m_TipoContatto;
                }
            }

            /// <summary>
            /// DataBlocco
            /// </summary>
            public DBCursorField<DateTime> DataBlocco
            {
                get
                {
                    return m_DataBlocco;
                }
            }

            /// <summary>
            /// IDBloccatoDa
            /// </summary>
            public DBCursorField<int> IDBloccatoDa
            {
                get
                {
                    return m_IDBloccatoDa;
                }
            }

            /// <summary>
            /// NomeBloccatoDa
            /// </summary>
            public DBCursorStringField NomeBloccatoDa
            {
                get
                {
                    return m_NomeBloccatoDa;
                }
            }

            /// <summary>
            /// MotivoBlocco
            /// </summary>
            public DBCursorStringField MotivoBlocco
            {
                get
                {
                    return m_MotivoBlocco;
                }
            }
             
            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.CustomerCalls.BlackListAdresses;
            }

            
        }
    }
}